src/mewbot/behaviour.py

```diff
@@ -2,70 +2,16 @@
 
 from __future__ import annotations
 
-from typing import Any, Dict, List, Optional, Set, Type
+from typing import Any, Dict, List, Set, Type, Union
 
-import abc
-
-from mewbot.component import Component
-from mewbot.io import InputEvent, OutputEvent, OutputQueue
-from mewbot.registry import limit_multi_inheritence
-
-
-class BehaviourComponent(Component):
-    @staticmethod
-    @abc.abstractmethod
-    def consumes_inputs() -> Set[Type[InputEvent]]:
-        pass
-
-
-@limit_multi_inheritence
-class Trigger(BehaviourComponent):
-    @abc.abstractmethod
-    def matches(self, event: InputEvent) -> bool:
-        pass
-
-
-@limit_multi_inheritence
-class Condition(BehaviourComponent):
-    @abc.abstractmethod
-    def allows(self, event: InputEvent) -> bool:
-        pass
-
-
-@limit_multi_inheritence
-class Action(BehaviourComponent):
-    @staticmethod
-    @abc.abstractmethod
-    def produces_outputs() -> Set[Type[OutputEvent]]:
-        pass
-
-    _queue: Optional[OutputQueue]
-
-    def __init__(self) -> None:
-        super().__init__()
-        self._queue = None
-
-    def bind(self, queue: OutputQueue) -> None:
-        self._queue = queue
-
-    def send(self, event: OutputEvent) -> None:
-        if not self._queue:
-            raise Exception("Can not sent events before queue initialisation")
-
-        self._queue.put_nowait(event)
-
-    @abc.abstractmethod
-    def act(self, event: InputEvent, state: Dict[str, Any]) -> None:
-        pass
+from mewbot.core import InputEvent, TriggerInterface, ConditionInterface, ActionInterface
 
 
 class Behaviour:
     active: bool = True
-    triggers: List[Trigger]  # This of things that can trigger this behaviour
-    conditions: List[
-        Condition
-    ]  # conditions on the action being called when it is matched (e.g. "is subscribed")
-    actions: List[Action]  # the things to do if an event matches, and the conditions are met
+    triggers: List[TriggerInterface]
```

Comment:
Minor/Readibility: Comments on each line were lost

Reply:
Intentional as inline comments should not be used for documentation (probably a hold over from copy pasting from the design documents?)
Currently pretty much everything is missing both doucmentation and tests